client/my-sites/site-settings/disconnect-site/index.jsx

```diff
@@ -9,28 +9,125 @@ import { localize } from 'i18n-calypso';
 /**
  * Internal dependencies
  */
+import { CompactCard } from '@automattic/components';
 import DisconnectSurvey from './disconnect-survey';
 import DocumentHead from 'components/data/document-head';
 import FormattedHeader from 'components/formatted-header';
+import Gridicon from 'components/gridicon';
 import Main from 'components/main';
 import NavigationLink from 'components/wizard/navigation-link';
 import redirectNonJetpack from 'my-sites/site-settings/redirect-non-jetpack';
 import Troubleshoot from './troubleshoot';
-import { getSelectedSiteSlug } from 'state/ui/selectors';
+import { getSelectedSiteSlug, getSelectedSite } from 'state/ui/selectors';
+import { localizeUrl } from 'lib/i18n-utils';
 
 /**
  * Style dependencies
  */
 import './style.scss';
 
-const DisconnectSite = ( { reason, siteSlug, translate } ) => {
-	const confirmHref = '/settings/disconnect-site/confirm/' + siteSlug;
+const DisconnectSite = ( {
+	reason,
+	type,
+	siteSlug,
+	site: { name: siteName, URL: siteUrl },
+	translate,
+} ) => {
+	let confirmHref = `/settings/disconnect-site/confirm/${ siteSlug }`;
 
 	let backHref = '/settings/manage-connection/' + siteSlug;
 	if ( reason ) {
 		backHref = '/settings/disconnect-site/' + siteSlug;
 	}
 
+	if ( type === 'down' ) {
```

Comment:
This big `if () { ... } else { ... }` block could be split into two React components.

Reply:
done in ae5d99c9a6cb2abeeb59f8daae2b175f203bcdae